src/chapters/thymeleaf-views/template-forms.rst

```diff
@@ -0,0 +1,152 @@
+piece of data you want to collect needs to have a unique ``name`` attribute.
+
+To *render* the form in the view, add a method to the controller using the
+``@GetMapping`` annotation:
+
+.. sourcecode:: Java
+   :linenos:
+
+   @GetMapping("formTemplateName")
+   public String renderFormMethodName(Model model) {
+
+      // Method code...
+
+      return "pathToTemplate";
+   }
+
+Some points to note:
+
+#. Line 1: The string parameter for ``GetMapping`` must be the name of the form
+   template you want to use.
+#. Line 2: Declare a ``Model`` object to hold data that needs to be passed to
+   the template.
+#. The method code performs any data manipulation required before rendering the
+   form. The ``model.addAttribute`` statements would be included here.
+#. The ``return`` string specifies the path to the template. Recall that Spring
+   automatically adds MOST of the file path---up through ``.../templates``. You
+   need to add any path details that follow.
+
+   a. For example, if our ``templates`` folder contains a subfolder called
+      ``events`` that holds a template called ``create.html``, then line 6
+      would be ``return "events/create";``.
+
+Add a Form Handler Method
+--------------------------
+
+Now that you have created and rendered a form in your ``coding-events``
+project, you need to add a method to the controller to *handle* its submission.
+Code along with the video below to add this functionality.
+
+.. todo:: ``Coding Events App: Process Add Event Form`` video goes here...
+
+As usual, the following summary outlines the ideas from the clip.
+
+Handle Form Submission
+^^^^^^^^^^^^^^^^^^^^^^^
+
+To *process* a form after the user clicks the *Submit* button, you need to add
+a method to the controller using the ``@PostMapping`` annotation:
+
+.. sourcecode:: Java
+   :linenos:
+
+   @PostMapping("formTemplateName")
+   public String processFormMethodName(@RequestParam Type parameter1, Type parameter2, ...) {
+
+      // Method code...
+
+      return "redirect:templateName";
+   }
+
+Some points to note:
+
+#. Line 1: The string parameter for ``PostMapping`` must be the name of the
+   form template.
+#. Line 2: For each piece of data that needs to be retrieved from the form,
+   declare a parameter of the appropriate type.
+
+   .. admonition:: Note
+
+      ``@RequestParam`` matches the parameters to the submitted data. For this
+      to work, the parameter names MUST match the ``name`` attributes used in
+      each of the ``input`` elements.
+
+#. The method code performs any data manipulation required after the
+   information gets submitted.
+#. Line 6: Generally, we want to send the user to a different page after they
+   successfully submit a form. Instead of re-rendering the form, the ``return``
+   string *redirects* the user to a method that handles a different template.
+
+Check Your Understanding
+-------------------------
+
+Questions go here...
```

Comment:
I don't know if there are really any good concept checks for this question since forms are covered in so much detail in other chapters.

Reply:
OK.  I removed this CC section for now.  If we think of a good one to add, we can always insert it later.